### ISSUE TITLE
Member type and media type compositions cannot be opened from where they are picked

### ISSUE THREAD

**Issue by alina-tincas** [umbraco/Umbraco-CMS#15009]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

tested on 12.1.0, 12.2.0 and 12.3.0-rc

### Bug summary

I get an error every time I try to open an inherited **member type or media type composition** from another member type or media type with the following error:
```
Request error: The URL returned a 404 (not found):
/umbraco/backoffice/umbracoapi/contenttype/GetById 

```
and in console:
```
Possibly unhandled rejection: {"errorMsg":"Failed to retrieve content type","data":{"Type":"https://tools.ietf.org/html/rfc7231#section-6.5.4","Title":"Not Found","Status":404,"Detail":null,"Instance":null,"Extensions":{"traceId":"00-fe571021c49db9372243c71ef2c9f799-48eb723654f9cdb6-00"}},"status":404,"xhrStatus":"complete"}
(anonymous) @ angular.js:15697
```

### Steps to reproduce

1. Create a new member type or media type 
2. Then choose a composition
3. Once a composition is chosen, click to open it and that is where you will get the mentioned errors.

You can also see attached a video of the steps:
https://github.com/umbraco/Umbraco-CMS/assets/83591955/4de8e2bb-d49a-4f41-a352-a684d323d04a

### Expected result / actual result

On a member or media type if you have a composition and wish to see it´s properties from where it is inherited it should open a window with this information instead of getting errors and having to refresh browser to get rid of the error.

**Comment by kjac** [comment]
Will investigate and report back here 👍

**Comment by kjac** [comment]
Fortunately this is "only" a client issue. The media and member types are saved correctly, and the composition setup can be applied to actual media and member instances.

The error occurs because the content type editor (which covers both content, media and members) seems hardcoded to look for content types when opening a composition in an infinite editor:

![image](https://github.com/umbraco/Umbraco-CMS/assets/7405322/bccd5cc4-b9d4-4edd-b342-f9526ccb0e5e)

The issue is also present in V11, so it's not quite a new thing.

I'm going to put this one up for grabs, if anyone wants to have a look. There are two ways to solve this:

1. Open the correct composition in the infinite editor. This is the preferred fix. It likely requires some contextual type handling to target the correct endpoint.
2. Disable the composition link for media and member types. While not ideal, this is definitively better than what we have now 😄

**Comment by kjac** [comment]
Thanks @rammi987 - and thank you for the fix ❤️

**Comment by kjac** [comment]
Fixed in #15017
